File: output/CodeLlama/codenet/Java/Python/s416486483.py
import sys

def split(n, k):
    sum = 0
    while n > 0:
        sum += n % k
        n /= k
    return sum

def solve(n, s):
    k = 2
    while k*k <= n or k < 100:
        if split(n, k) == s:
            return k
        k += 1
    while k <= n:
        a = n / k
        b = n % k
        if (a + b - s) % a == 0:
            dk = (a + b - s) / a
            if dk >= 0 and b - dk*a >= 0:
                return k + dk
        k = n / a + 1
    if n == s:
        return n+1
    return -1

if __name__ == '__main__':
    nl = int(sys.stdin.readline())
    sl = int(sys.stdin.readline())
    print(solve(nl, sl))

File: output/ChatGPT/codenet/Java/C++/s416486483.cpp
#include <iostream>

using namespace std;

long split(long n, long k) {
  long sum = 0;
  while (n > 0) {
    sum += n % k;
    n /= k;
  }
  return sum;
}

long solve(long n, long s) {
  long k;
  for (k = 2; k*k <= n || k < 100; k++) {
    if (split(n, k) == s) {
      return k;
    }
  }
  while (k <= n) {
    long a = n / k;
    long b = n % k;
    if ((a + b - s) % a == 0) {
      long dk = (a + b - s) / a;
      if (dk >= 0 && b - dk*a >= 0) {
        return k + dk;
      }
    }
    k = n / a + 1;
  }
  if (n == s) {
    return n+1;
  }
  return -1;
}

int main() {
  long nl, sl;
  cin >> nl >> sl;
  cout << solve(nl, sl) << endl;
}